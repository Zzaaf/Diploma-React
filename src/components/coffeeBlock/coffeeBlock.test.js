import React from 'react';
import {shallow} from 'enzyme';
import CoffeeBlock from '../coffeeBlock';
import CoffeeService from '../../services/coffeeService';

describe('Testing <CoffeeBlock/>', () => {
    const service = new CoffeeService();
    const items = shallow(<CoffeeBlock/>);
    describe('Testing snap & state', () => {
        it('CoffeeBlock has rendered correctly', () => {
            expect(items).toMatchSnapshot();
        });
        it('CoffeeBlock state "posts" is array', () => {
            expect(items.state().posts).toBeArray();
        });
        it('CoffeeBlock state "typeError" is string', () => {
            expect(items.state().typeError).toBeString();
        });
        it('CoffeeBlock state "fatalError" is false', () => {
            expect(items.state().fatalError).toBeFalsy();
        });
    });
});