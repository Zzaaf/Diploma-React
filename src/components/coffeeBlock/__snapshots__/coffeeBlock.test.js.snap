// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CoffeeBlock/> Testing snap & state CoffeeBlock has rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CoffeeBlock />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Row
          tag="div"
        >
          <div
            className="col-lg-4 offset-2"
          >
            <form
              action="#"
              className="shop__search"
            >
              <label
                className="shop__search-label"
                htmlFor="filter"
              >
                Looking for
              </label>
              <input
                className="shop__search-input"
                id="filter"
                onChange={[Function]}
                placeholder="start typing here..."
                type="text"
              />
            </form>
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="shop__filter"
            >
              <div
                className="shop__filter-label"
              >
                Or filter
              </div>
              <div
                className="shop__filter-group"
              >
                <button
                  className="shop__filter-btn"
                  onClick={[Function]}
                >
                  Brazil
                </button>
                <button
                  className="shop__filter-btn"
                  onClick={[Function]}
                >
                  Kenya
                </button>
                <button
                  className="shop__filter-btn"
                  onClick={[Function]}
                >
                  Columbia
                </button>
              </div>
            </div>
          </div>
        </Row>,
        <CoffeeList
          onCoffeeSelected={[Function]}
          posts={Array []}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="col-lg-4 offset-2"
            >
              <form
                action="#"
                className="shop__search"
              >
                <label
                  className="shop__search-label"
                  htmlFor="filter"
                >
                  Looking for
                </label>
                <input
                  className="shop__search-input"
                  id="filter"
                  onChange={[Function]}
                  placeholder="start typing here..."
                  type="text"
                />
              </form>
            </div>,
            <div
              className="col-lg-4"
            >
              <div
                className="shop__filter"
              >
                <div
                  className="shop__filter-label"
                >
                  Or filter
                </div>
                <div
                  className="shop__filter-group"
                >
                  <button
                    className="shop__filter-btn"
                    onClick={[Function]}
                  >
                    Brazil
                  </button>
                  <button
                    className="shop__filter-btn"
                    onClick={[Function]}
                  >
                    Kenya
                  </button>
                  <button
                    className="shop__filter-btn"
                    onClick={[Function]}
                  >
                    Columbia
                  </button>
                </div>
              </div>
            </div>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                action="#"
                className="shop__search"
              >
                <label
                  className="shop__search-label"
                  htmlFor="filter"
                >
                  Looking for
                </label>
                <input
                  className="shop__search-input"
                  id="filter"
                  onChange={[Function]}
                  placeholder="start typing here..."
                  type="text"
                />
              </form>,
              "className": "col-lg-4 offset-2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "action": "#",
                "children": Array [
                  <label
                    className="shop__search-label"
                    htmlFor="filter"
                  >
                    Looking for
                  </label>,
                  <input
                    className="shop__search-input"
                    id="filter"
                    onChange={[Function]}
                    placeholder="start typing here..."
                    type="text"
                  />,
                ],
                "className": "shop__search",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Looking for",
                    "className": "shop__search-label",
                    "htmlFor": "filter",
                  },
                  "ref": null,
                  "rendered": "Looking for",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "shop__search-input",
                    "id": "filter",
                    "onChange": [Function],
                    "placeholder": "start typing here...",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "form",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="shop__filter"
              >
                <div
                  className="shop__filter-label"
                >
                  Or filter
                </div>
                <div
                  className="shop__filter-group"
                >
                  <button
                    className="shop__filter-btn"
                    onClick={[Function]}
                  >
                    Brazil
                  </button>
                  <button
                    className="shop__filter-btn"
                    onClick={[Function]}
                  >
                    Kenya
                  </button>
                  <button
                    className="shop__filter-btn"
                    onClick={[Function]}
                  >
                    Columbia
                  </button>
                </div>
              </div>,
              "className": "col-lg-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="shop__filter-label"
                  >
                    Or filter
                  </div>,
                  <div
                    className="shop__filter-group"
                  >
                    <button
                      className="shop__filter-btn"
                      onClick={[Function]}
                    >
                      Brazil
                    </button>
                    <button
                      className="shop__filter-btn"
                      onClick={[Function]}
                    >
                      Kenya
                    </button>
                    <button
                      className="shop__filter-btn"
                      onClick={[Function]}
                    >
                      Columbia
                    </button>
                  </div>,
                ],
                "className": "shop__filter",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Or filter",
                    "className": "shop__filter-label",
                  },
                  "ref": null,
                  "rendered": "Or filter",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <button
                        className="shop__filter-btn"
                        onClick={[Function]}
                      >
                        Brazil
                      </button>,
                      <button
                        className="shop__filter-btn"
                        onClick={[Function]}
                      >
                        Kenya
                      </button>,
                      <button
                        className="shop__filter-btn"
                        onClick={[Function]}
                      >
                        Columbia
                      </button>,
                    ],
                    "className": "shop__filter-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "0",
                      "nodeType": "host",
                      "props": Object {
                        "children": "Brazil",
                        "className": "shop__filter-btn",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Brazil",
                      "type": "button",
                    },
                    Object {
                      "instance": null,
                      "key": "1",
                      "nodeType": "host",
                      "props": Object {
                        "children": "Kenya",
                        "className": "shop__filter-btn",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Kenya",
                      "type": "button",
                    },
                    Object {
                      "instance": null,
                      "key": "2",
                      "nodeType": "host",
                      "props": Object {
                        "children": "Columbia",
                        "className": "shop__filter-btn",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Columbia",
                      "type": "button",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onCoffeeSelected": [Function],
          "posts": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Row
            tag="div"
          >
            <div
              className="col-lg-4 offset-2"
            >
              <form
                action="#"
                className="shop__search"
              >
                <label
                  className="shop__search-label"
                  htmlFor="filter"
                >
                  Looking for
                </label>
                <input
                  className="shop__search-input"
                  id="filter"
                  onChange={[Function]}
                  placeholder="start typing here..."
                  type="text"
                />
              </form>
            </div>
            <div
              className="col-lg-4"
            >
              <div
                className="shop__filter"
              >
                <div
                  className="shop__filter-label"
                >
                  Or filter
                </div>
                <div
                  className="shop__filter-group"
                >
                  <button
                    className="shop__filter-btn"
                    onClick={[Function]}
                  >
                    Brazil
                  </button>
                  <button
                    className="shop__filter-btn"
                    onClick={[Function]}
                  >
                    Kenya
                  </button>
                  <button
                    className="shop__filter-btn"
                    onClick={[Function]}
                  >
                    Columbia
                  </button>
                </div>
              </div>
            </div>
          </Row>,
          <CoffeeList
            onCoffeeSelected={[Function]}
            posts={Array []}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                className="col-lg-4 offset-2"
              >
                <form
                  action="#"
                  className="shop__search"
                >
                  <label
                    className="shop__search-label"
                    htmlFor="filter"
                  >
                    Looking for
                  </label>
                  <input
                    className="shop__search-input"
                    id="filter"
                    onChange={[Function]}
                    placeholder="start typing here..."
                    type="text"
                  />
                </form>
              </div>,
              <div
                className="col-lg-4"
              >
                <div
                  className="shop__filter"
                >
                  <div
                    className="shop__filter-label"
                  >
                    Or filter
                  </div>
                  <div
                    className="shop__filter-group"
                  >
                    <button
                      className="shop__filter-btn"
                      onClick={[Function]}
                    >
                      Brazil
                    </button>
                    <button
                      className="shop__filter-btn"
                      onClick={[Function]}
                    >
                      Kenya
                    </button>
                    <button
                      className="shop__filter-btn"
                      onClick={[Function]}
                    >
                      Columbia
                    </button>
                  </div>
                </div>
              </div>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <form
                  action="#"
                  className="shop__search"
                >
                  <label
                    className="shop__search-label"
                    htmlFor="filter"
                  >
                    Looking for
                  </label>
                  <input
                    className="shop__search-input"
                    id="filter"
                    onChange={[Function]}
                    placeholder="start typing here..."
                    type="text"
                  />
                </form>,
                "className": "col-lg-4 offset-2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "action": "#",
                  "children": Array [
                    <label
                      className="shop__search-label"
                      htmlFor="filter"
                    >
                      Looking for
                    </label>,
                    <input
                      className="shop__search-input"
                      id="filter"
                      onChange={[Function]}
                      placeholder="start typing here..."
                      type="text"
                    />,
                  ],
                  "className": "shop__search",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Looking for",
                      "className": "shop__search-label",
                      "htmlFor": "filter",
                    },
                    "ref": null,
                    "rendered": "Looking for",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "shop__search-input",
                      "id": "filter",
                      "onChange": [Function],
                      "placeholder": "start typing here...",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "form",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="shop__filter"
                >
                  <div
                    className="shop__filter-label"
                  >
                    Or filter
                  </div>
                  <div
                    className="shop__filter-group"
                  >
                    <button
                      className="shop__filter-btn"
                      onClick={[Function]}
                    >
                      Brazil
                    </button>
                    <button
                      className="shop__filter-btn"
                      onClick={[Function]}
                    >
                      Kenya
                    </button>
                    <button
                      className="shop__filter-btn"
                      onClick={[Function]}
                    >
                      Columbia
                    </button>
                  </div>
                </div>,
                "className": "col-lg-4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="shop__filter-label"
                    >
                      Or filter
                    </div>,
                    <div
                      className="shop__filter-group"
                    >
                      <button
                        className="shop__filter-btn"
                        onClick={[Function]}
                      >
                        Brazil
                      </button>
                      <button
                        className="shop__filter-btn"
                        onClick={[Function]}
                      >
                        Kenya
                      </button>
                      <button
                        className="shop__filter-btn"
                        onClick={[Function]}
                      >
                        Columbia
                      </button>
                    </div>,
                  ],
                  "className": "shop__filter",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Or filter",
                      "className": "shop__filter-label",
                    },
                    "ref": null,
                    "rendered": "Or filter",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <button
                          className="shop__filter-btn"
                          onClick={[Function]}
                        >
                          Brazil
                        </button>,
                        <button
                          className="shop__filter-btn"
                          onClick={[Function]}
                        >
                          Kenya
                        </button>,
                        <button
                          className="shop__filter-btn"
                          onClick={[Function]}
                        >
                          Columbia
                        </button>,
                      ],
                      "className": "shop__filter-group",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": "0",
                        "nodeType": "host",
                        "props": Object {
                          "children": "Brazil",
                          "className": "shop__filter-btn",
                          "onClick": [Function],
                        },
                        "ref": null,
                        "rendered": "Brazil",
                        "type": "button",
                      },
                      Object {
                        "instance": null,
                        "key": "1",
                        "nodeType": "host",
                        "props": Object {
                          "children": "Kenya",
                          "className": "shop__filter-btn",
                          "onClick": [Function],
                        },
                        "ref": null,
                        "rendered": "Kenya",
                        "type": "button",
                      },
                      Object {
                        "instance": null,
                        "key": "2",
                        "nodeType": "host",
                        "props": Object {
                          "children": "Columbia",
                          "className": "shop__filter-btn",
                          "onClick": [Function],
                        },
                        "ref": null,
                        "rendered": "Columbia",
                        "type": "button",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onCoffeeSelected": [Function],
            "posts": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
